### ISSUE TITLE
Support netlink connector sockets in System.Net.Sockets

### ISSUE THREAD

**Issue by divman25** [dotnet/corefx#34015]
I am unable to use netlink connectors on Linux via dotnet and socket creation itself fails with 

> Unhandled Exception: System.Net.Sockets.SocketException: Address family not supported by protocol

I am trying to use linux netlink connectors to listen to process fork events directly from the OS. The C code would look something like this to create a socket and then use **sockaddr_nl** to send messages- 
`socket(PF_NETLINK, SOCK_DGRAM, NETLINK_CONNECTOR)`
OR
```
int main() {
    int sock = socket(16, SOCK_DGRAM, 11);
    return 0;
}
```
While the corresponding C# fails - 
```
const UInt16 NETLINK_CONNECTOR = 11;
const UInt16 PF_NETLINK = 16;
Socket temp = new Socket((AddressFamily)PF_NETLINK, SocketType.Dgram, (ProtocolType)NETLINK_CONNECTOR);
```

**Comment by karelz** [comment]
cc @tmds @wfurt @eerhardt

**Comment by wfurt** [comment]
This will need API review. We talk about it when we talk about AF_PACKET. This one is also only Linux specific e.g. no windows, no other UNIXes.

**Comment by karelz** [comment]
I assume you meant related to #30197

**Comment by wfurt** [comment]
yes, I should have added the link. In both cases we need to add enum to protocol list and do some testing on _all_ platforms. Since this would work only on Linux, we may do some extra documentation to make it clear.

**Comment by karelz** [comment]
Triage: Does not seem critical, but valuable.
